qiskit/_quantumcircuit.py

```diff
@@ -37,6 +37,7 @@ class QuantumCircuit(object):
 
     def __init__(self, *regs):
         """Create a new circuit."""
+        self.name = None
```

Comment:
Because the name is not given in `QuantumCircuit` construction time, but when the circuit is added to the quantum program. Yeap, I dont like that the circuits dont have a name per se neither. But I think in this way the original approach is kept (even when I'm not agreeing with it).

Reply:
But if you are adding a `name` attribute to the class, you are effectively saying that "the circuits have a name" - and arguably modifying the original approach. Can you check the real impact of adding it to the constructor (as far as I can see, outside of the SDK there is only one tutorial that would need to be updated)? I feel that we should either modify the class completely, or not modify it at all, but standing in middle ground is not ideal (`QuantumCircuit.name` becomes a variable that is not really set by the class itself).

Reply:
done!